front-end/src/temperature/readings_chart.jsx

```diff
@@ -41,9 +41,9 @@ class chart extends React.Component {
                 <stop offset='95%' stopColor={c.color} stopOpacity={0} />
               </linearGradient>
             </defs>
-            <YAxis dataKey='value' domain={[c.ymin, c.ymax]} />
+            <YAxis dataKey='value' allowDecimals='false' domain={[c.ymin, c.ymax]} />
             <XAxis dataKey='time' />
-            <Tooltip formatter={(value, name) => [value, unit]} />
+            <Tooltip formatter={(value, name) => [parseFloat(value).toFixed(2), unit]} />
```

Comment:
toFixed will cause error if parseFloat returns NaN. Either omit this or add a helper method  for two decimal precision instead of directly calling toFixed that can deal with parseFloat failures

Reply:
Added utility function two_decimal_parse.js to handle parseFloat failures.